### PR TITLE
AxisManager utility functions

### PR DESCRIPTION
**Pull request by jlashner**
A couple of useful axismanager functions implemented in sodetlib usage. Katie said a few may be more generally applicable so I'm copying them here.

If these are merged in I'll remove them from sodetlib.

### CODE REVIEW

File: sotodlib/core/axisman_util.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+"""
+Various AxisManager utility functions
+"""
+
+import numpy as np
+from sotodlib.core import AxisManager
+
+class RestrictionException(Exception):
+    """Exception for when cannot restrict AxisManger properly"""
```
Comment (mhasself) [inline sotodlib/core/axisman_util.py:9]: AxisManager.

**Review by mhasself — CHANGES_REQUESTED** [review]
One could see either of these getting into the class itself (restrict_to_times would need to be a bit more general; restrict_by_value or something).  But good to have them in here.

Please add basic smoke testing for these to tests/test_core.py

**Comment by jlashner** [comment]
Thanks for the review! Comments implemented and tests added

**Review by mhasself — APPROVED** [review]
Sorry for the slow follow-up.  Thanks for the changes.

**Pull request merged**